discovery_engine_core/web-api2/src/error/application.rs

```diff
@@ -105,11 +105,11 @@ macro_rules! impl_application_error {
                 stringify!($name)
             }
 
-            fn encode_details(&self) -> serde_json::Value {
+            fn encode_details(&self) -> ::serde_json::Value {
                 ::serde_json::to_value(self)
                     .unwrap_or_else(|err| {
-                        error!(%err, "serializing error details failed");
-                        serde_json::Value::Null
+                        ::tracing::error!(%err, "serializing error details failed");
+                        ::serde_json::Value::Null
```

Comment:
❓ why the additional `::` qualifier? it's quite unlikely that we confuse the external crate with some internal path. even if we decide to take this precautionary measure, it then should be applied everywhere in the code, which should be done in a separate pr.

Reply:
Just general best practice for macros with `#[macro_export]`,  I don't think we should bother applying it everywhere in the code it doesn't add any value until it causes a problem, at which point we can change it. But there is no reason to not use it here.